tests/test_self_tools.py

```
@@ -34,10 +34,24 @@ def setUp(self) -> None:
+                shutil.rmtree(self.temp_dir)
+            except Exception as e:
+                print(f"Warning: Could not clean up temp directory: {e}")
+
+        # Safety cleanup: remove any test directories that leaked into CWD
+        for dirname in ["dir1", "dir2", "dir3", "sub1_1", "sub1_2", "sub2_1", "sub2_2"]:
+            dir_path = os.path.join(current_cwd, dirname)
+            if os.path.exists(dir_path) and os.path.isdir(dir_path):
+                try:
+                    shutil.rmtree(dir_path, ignore_errors=True)
+                except Exception:
+                    pass
```

Ruff (0.13.3)

`try`-`except`-`pass` detected, consider logging the exception

(S110)